### DISCUSSION TITLE
How to change Characters between (P1 - IA) (sorry for my bad english)

### DISCUSSION THREAD

**Discussion by The-Enazer** [Techokami/SonicWorldsNext#210] [Q&A]
Hello, Well you see, I make you can choose the, Playerchar1 and Playerchar2. (player and ia) But, I don't know how to make it so that with one key you can change from one to the other. Could you explain to me?
Thanks for everything (Example: Sonic Triple Trouble 16 bits change of sonic and tails)

**Comment by Raptorq1** [comment]
> Hello, Well you see, I make you can choose the, Playerchar1 and Playerchar2. (player and ia) But, I don't know how to make it so that with one key you can change from one to the other. Could you explain to me? Thanks for everything (Example: Sonic Triple Trouble 16 bits change of sonic and tails)

You forgot to reference mania plus

**Reply by The-Enazer** [reply]
yeah i guess. lmao

**Reply by The-Enazer** [reply]
No, in sonic mania plus you change the character in the encore mode. I, gonna make you can change between playerchar1 and playerchar2.

**Reply by Raptorq1** [reply]
i know that Its still the same mechanic

**Comment by NeonSRB2** [comment]
Idk if this goes through email or smth... Writing this so I can paste it to
the thread later.

Since you have full control over what the player is... You can just change
it There should be an array or something of players, swap those around so
player 1 is player 2 and player 2 is player 1. (Make sure you do keep the
value of the first variable you change, so you don't lose player 1's node
completely!
Swap that bot variable
Swap the camera variable

Both games you mentioned put restrictions on when you could swap for some
reason. I don't know how these work, and I don't like them either.

**Reply by NeonSRB2** [reply]
oh shoot it vvorks

**Reply by NeonSRB2** [reply]
ill add code snippets later

**Reply by Raptorq1** [reply]
> ill add code snippets later

Where's the code

**Reply by The-Enazer** [reply]
thanks il be wait for the code

**Reply by Raptorq1** [reply]
@NeonSRB2 can you add the code

**Reply by Raptorq1** [reply]
Hey @NeonSRB2 please add the code

**Reply by r6915ee** [reply]
They haven't been really active for a bit, Raptor. Give them some time.

**Reply by Raptorq1** [reply]
Okay

**Reply by Raptorq1** [reply]
@NeonSRB2 I followed your tutorial and got it to work and but the characters are not changing only the abilities are changing

if Input.is_action_pressed("gm_action2"):
			character = char.pick_random()
			partner.character = char.pick_random()
			respawn()

**Reply by DimensionWarped** [reply]
> Idk if this goes through email or smth... Writing this so I can paste it to the thread later. Since you have full control over what the player is... You can just change it There should be an array or something of players, swap those around so player 1 is player 2 and player 2 is player 1. (Make sure you do keep the value of the first variable you change, so you don't lose player 1's node completely! Swap that bot variable Swap the camera variable Both games you mentioned put restrictions on when you could swap for some reason. I don't know how these work, and I don't like them either.
> […](#)
> On Sat, Nov 2, 2024, 5:36 PM Raptorkillz12 ***@***.***> wrote: i know that Its still the same mechanic — Reply to this email directly, view it on GitHub <[#210 (reply in thread)](https://github.com/Techokami/SonicWorldsNext/discussions/210#discussioncomment-11131488)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/A4JOMPPID5LZCYXGYDAI5HLZ6VAU5AVCNFSM6AAAAABQ7IO4SKVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMJTGE2DQOA> . You are receiving this because you are subscribed to this thread.Message ID: <Techokami/SonicWorldsNext/repo-discussions/210/comments/11131488@ github.com>

Only restriction I'd want to put on it is that both characters most be on the screen.

I have a feeling aspects of this are going to be slightly annoying in hard to foresee ways.

**Reply by Raptorq1** [reply]
@NeonSRB2 Can you add the code please
